helpers.sh

```diff
@@ -75,29 +75,26 @@ function repo_sync {
 }
 
 function set_base_lmp_version {
-	# Detect the base LmP version we are building on
-	# A user may have place arbitrary tags into their lmp-manifest and we
-	# don't want to accidentally pick one of them up. Git doesn't really
-	# distinguish where tags come from, so:
+	# 1: Work on a copy
+	WORKCOPY="$PWD/.repo/manifests.git.tmp"
+	cp -R .repo/manifests.git $WORKCOPY
+	pushd $WORKCOPY
 
-	# 1: Delete all tags
-	git --git-dir .repo/manifests.git tag | xargs git --git-dir .repo/manifests.git tag -d > /dev/null
-
-	# 2: Get the LMP tags
-	run git --git-dir .repo/manifests.git remote add upstream https://github.com/foundriesio/lmp-manifest
-	run git --git-dir .repo/manifests.git fetch --tags upstream
+	# 2: Replace all tags
+	run git tag --delete $(git tag) >/dev/null
+	run git remote set-url origin https://github.com/foundriesio/lmp-manifest
+	run git fetch origin --tags --quiet
 
 	# 3: Find our base LMP version based on the HEAD
-	export LMP_VER=$(git --git-dir .repo/manifests.git describe HEAD --tags --abbrev=0)
+	export LMP_VERSION=$(git describe --tags HEAD)
```

Comment:
I wonder if it would just be better to also fix LMP_VERSION to use abbrev=0 as well, making it the same as LMP_CACHE_VERSION.

Since what we really want here is the tag revision.

Reply:
we need to use two variables for this because the version used on the sstate-cache can be incremented.

Reply:
While now it is easy to understand, I find reusing these variables a bit confusing and I'm sure it will cause confusion later on once we have to change this codebase again.

Why don't we just using one single LMP_VERSION variable (fetching with abbrev=0) and just define LMP_VERSION_CACHE instead of VERSION_NUM (which can be confusing).

That way we set LMP_VERSION as the global variable to be used to find out the version the build is based on, and LMP_VERSION_CACHE used to point to the cache version/bucket to be used by the build instead.

Reply:
Change LMP_VERSION variable (fetching with abbrev=0) can impact the target roofs as it will change `os-release`, is this acceptable? 

We can maintain the LMP_VERSION as before (i.e `89-27-g8f931b6`) and strip down only the release number with cut, storing the result on  LMP_VERSION_CACHE that can be incremented and will be used for sstate cache.
```
# LMP_VERSION=89-27-g8f931b6
LMP_VERSION_CACHE="$(echo $LMP_VERSION | cut -d'-' -f1)"
```

Reply:
That is fine as every CI job will have a customized version field as it contains the build number as well.

Reply:
And my suggestion is because we should have used abbrev=0 when we did this first implementation.

Reply:
Fixed